src/polynomial_optimizer.py

```diff
@@ -0,0 +1,232 @@
+import numpy as np
```

Comment:
Please add some basic comments to the top of this explaining what it's for (eg the coefficients for fast_atan2 etc), when someone might want to use it in the future, and how to use it if so